ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JClient.java

```diff
@@ -77,39 +82,50 @@ public <T> SafeFuture<Response<T>> doRequest(
             (response, exception) -> {
               if (exception != null) {
                 final boolean couldBeAuthError = isAuthenticationException(exception);
-                handleError(exception, couldBeAuthError);
+                handleError(isCriticalRequest(web3jRequest), exception, couldBeAuthError);
```

Comment:
I would initialise beforehand `boolean isCritical = isCriticalRequest(web3jRequest)` since it is used in all calls